web/src/recent_view_ui.ts

```diff
@@ -1133,8 +1140,13 @@ function recenter_focus_if_off_screen(): void {
         const topic_center_y = (position.top + position.bottom) / 2;
 
         const topic_element = document.elementFromPoint(topic_center_x, topic_center_y);
-        assert(topic_element !== null);
-        row_focus = $topic_rows.index($(topic_element).closest("tr")[0]);
+        if (topic_element === null) {
+            // Last visible element is above the vertical center of the table.
```

Comment:
Can you spell out a bit more what this means / why it is correct? I'm missing some detail here.

Reply:
So, there are 2 possibilities of `topic_element` being null here. 

* One is if we haven't rendered the list yet, which is fixed by is_waiting_for_revive_current_focus which fixes the sentry error.
* Other is if there is no row at the vertical center of the recent view but slightly above it. In that case, we focus on the last element. This is just something that I noticed is possible but not entierly sure if it is possible to get into this state.

Reply:
OK I wrote a comment trying to explain this and will merge ... but I'm not sure it's entirely correct.

Also, `$topic_rows.index($(topic_element).closest("tr")[0]);` seems like it's not limited to just conversation rows, but it could fire on a header row : I wonder if we shouldn't be using `table_wrapper_element.getBoundingClientRect();`, but instead `table_wrapper_element.find("tbody").getBoundingClientRect();`... I guess `find` is bad because of nested tables, but you get the idea.